clang/lib/Driver/Driver.cpp

```diff
@@ -4791,6 +4791,20 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     if (!UnbundlerInputs.empty()) {
       Action *PartialLink =
           C.MakeAction<PartialLinkJobAction>(UnbundlerInputs, types::TY_Object);
+
+      auto *tAC = dyn_cast<Action>(UnbundlerInputs.back());
+      assert(tAC->getKind() == Action::InputClass ||
+             tAC->getKind() == Action::OffloadClass);
+
+      // if the Action is of OffloadAction type, use the HostAction of the
+      // specfic Offload Action to set LastArg
+      if (auto *IA = dyn_cast<OffloadAction>(UnbundlerInputs.back()))
+        LastArg =
+            &(dyn_cast<InputAction>(IA->getHostDependence())->getInputArg());
+      else if (auto *IA = dyn_cast<InputAction>(UnbundlerInputs.back()))
+        // else set the LastArg based on Last InputAction
+        LastArg = &(IA->getInputArg());
+
```

Comment:
somethings missing here...  You are overwriting LastArg with the UnbundlerInput even if LastArg was populated.  This should only be done when LastArg hasn't been set.